pydatastructs/graphs/algorithms.py

```diff
@@ -548,3 +550,64 @@ def strongly_connected_components(graph, algorithm):
         "isn't implemented for finding strongly connected components."
         %(algorithm, graph._impl))
     return getattr(algorithms, func)(graph)
+
+
+def random_walk(graph, start, path_length, seed = 0, alpha = None):
+    """
+    Executes random walk from the starting node and given graph.
+
+    Parameters
+    ==========
+
+    graph: Graph
+        The graph on which random walk needs to be executed.
+    start: GraphNode or it's subclasses
+        The node/vertex from where random walk will begin.
+    path_length:
+        The maximum number of nodes/vertices traversed by random walk.
+    alpha:
+        A parameter which determines the randomness. More the value more the chances of restart.
+
+    Returns
+    =======
+
+    path: list
+        Python list of nodes after execution of random walk.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import Graph, AdjacencyListGraphNode
+    >>> from pydatastructs import random_walk
+    >>> v1, v2, v3, v4, v5  = [AdjacencyListGraphNode(i) for i in range(5)]
+    >>> g = Graph(v1, v2, v3, v4, v5)
+    >>> g.add_edge(v1.name, v2.name)
+    >>> g.add_edge(v2.name, v3.name)
+    >>> g.add_edge(v3.name, v1.name)
+    >>> g.add_edge(v2.name, v4.name)
+    >>> g.add_edge(v4.name, v5.name)
+
+    >>> random_walk = random_walk(g, v1, 3)
+    >>> random_walk == ['v1', 'v2', 'v5']
+    True
+
+    """
+
+    path = [start.name]
+
+    for i in range(0, path_length):
+        cur = path[-1]
+        degree = len(graph.neighbors(cur))
+        probablity_list = []
+
+        if alpha is not None:
+            if random.random() < alpha:
+                next_node = start.name
+        else:
+            if graph.neighbors(cur) == []:
+                break
+            next_node = random.choices(graph.neighbors(cur))
```

Comment:
I am not sure, whether `random.choices` is truly uniform. A simple empirical analysis will help figuring that thing out.

Reply:
There is one more parameter of weights, which is probability list, I removed it while debugging, I will put it back.

Reply:
If `random.choices` picks uniformly from the list, then may be we do not need `probability_list`. 


> A simple empirical analysis will help figuring that thing out.

Reply:
```python
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> l = ["hey", "wassup", "byeeee", "timesupppppppppppp"]
>>> import random
>>> sample = [random.choices(l) for _ in range(1000)]
>>> sample.count(['hey'])
234
>>> sample.count(['wassup'])
257
>>> sample.count(['byeeee'])
236
>>> sample.count(['timesupppppppppppp'])
273
>>> sample = [random.choices(l) for _ in range(100000)]
>>> sample.count(['hey'])
24903
>>> sample.count(['wassup'])
24945
>>> sample.count(['byeeee'])
25246
>>> sample.count(['timesupppppppppppp'])
24906
>>> sample = [random.choices(l) for _ in range(1000000)]
>>> sample.count(['hey'])
249375
>>> sample.count(['wassup'])
250203
>>> sample.count(['byeeee'])
250014
>>> sample.count(['timesupppppppppppp'])
250408
>>> l = [1, 2, 3, -3, -2, -1]
>>> sample = [random.choices(l) for _ in range(1000000)]
>>> [sample.count([i]) for i in l]
[166668, 167257, 166685, 166324, 166541, 166525]
>>> 1000000/6
166666.66666666666
```
The above shows that, `random.choices` uniformly selects if `wieghts` is `None`. So no need to waste CPU cycles on computing `probability_list` or `degree`.